jcl/src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -529,8 +529,59 @@ public StringBuffer append (float value) {
  * @param		value	the integer
  * @return		this StringBuffer
  */
-public StringBuffer append (int value) {
+public synchronized StringBuffer append(int value) {
+	/*[IF Sidecar19-SE]*/
+	/*[IF Sidecar19-SE-B174]*/
+	int currentLength = lengthInternalUnsynchronized();
+	int currentCapacity = capacityInternal();
+
+	int valueLength = Integer.stringSize(value);
+
+	int newLength = currentLength + valueLength;
+
+	if (newLength > currentCapacity) {
+		ensureCapacityImpl(newLength);
+	}
+
+	if (String.enableCompression && count >= 0) {
+		Integer.getChars(value, newLength, this.value);
+	} else {
+		StringUTF16.getChars(value, newLength, this.value);
+	}
+
+	count = newLength;
```

Comment:
This may change the sign of `count` - is that intentional?

Reply:
This is a valid issue and a mistake. Fixed in a68ca65